impersonate/impersonate.go

```diff
@@ -8,20 +8,25 @@ import (
 	"bytes"
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"net/http"
 	"time"
 
 	"golang.org/x/oauth2"
+	"google.golang.org/api/internal"
 	"google.golang.org/api/option"
 	"google.golang.org/api/option/internaloption"
 	htransport "google.golang.org/api/transport/http"
 )
 
 var (
-	iamCredentailsEndpoint = "https://iamcredentials.googleapis.com"
-	oauth2Endpoint         = "https://oauth2.googleapis.com"
+	iamCredentailsEndpoint                      = "https://iamcredentials.googleapis.com"
+	oauth2Endpoint                              = "https://oauth2.googleapis.com"
+	ErrUniverseNotSupportedDomainWideDelegation = errors.New(
```

Comment:
I don't think this needs to be a public error. There is no action for the user to take programmatically if this error occurs.

Reply:
Same with other errors

Reply:
That's reasonable feedback.  If we end up using the error in multiple places it might make sense to keep it but leave it unexported.

Reply:
Fixed, thanks! Also improved the impersonate error unit tests.